File: Codes/2962-count-subarrays-where-max-element-appears-at-least-k-times.py
'''
Author: LetMeFly
Date: 2025-04-29 13:19:47
LastEditors: LetMeFly.xyz
LastEditTime: 2025-04-29 21:21:03
'''
from typing import List

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        mx = max(nums)
        l = cnt = ans = 0
        for t in nums:
            cnt += t == mx
            while cnt == k:
                cnt -= nums[l] == mx
                l += 1
            ans += l
        return ans

File: Codes/2962-count-subarrays-where-max-element-appears-at-least-k-times.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-04-29 13:19:27
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-29 21:19:24
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

typedef long long ll;

class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int mx = nums[0];
        for (int t : nums) {
            mx = max(mx, t);
        }
        ll ans = 0;
        for (int l = 0, r = 0, cnt = 0; r < nums.size(); r++) {
            cnt += nums[r] == mx;
            while (cnt == k) {
                cnt -= nums[l++] == mx;
            }
            ans += l;
        }
        return ans;
    }
};